raft/raft.go

```diff
@@ -547,13 +548,23 @@ func (r *raft) bcastHeartbeatWithCtx(ctx []byte) {
 // the commit index changed (in which case the caller should call
 // r.bcastAppend).
 func (r *raft) maybeCommit() bool {
-	// TODO(bmizerany): optimize.. Currently naive
-	mis := make(uint64Slice, 0, len(r.prs))
+	if cap(r.matchBuf) < len(r.prs) {
```

Comment:
probably document this optimization for avoiding allocations on slice creating and sorting.

Reply:
Done.